src/main/java/com/cloudbees/jenkins/plugins/bitbucket/server/client/BitbucketServerAPIClient.java

```diff
@@ -481,12 +488,29 @@ private String getRequest(String path) throws IOException {
         HttpClient client = getHttpClient(getMethodHost(httpget));
         try {
             client.executeMethod(httpget);
-            String response = new String(httpget.getResponseBody(), "UTF-8");
+            String response;
+            long len = httpget.getResponseContentLength();
```

Comment:
Would require more aggressive refactoring as there are subtle method signature changes that need ironing out and without good test coverage risk breaking things